assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java

```diff
@@ -292,6 +293,9 @@ protected void configure() {
       install(new org.eclipse.che.core.metrics.MetricsModule());
       install(new WsMasterMetricsModule());
     }
+
+    // now that everything is installed, we can recover the state of the workspaces
+    getProvider(WorkspaceManager.class).get().recover();
```

Comment:
Could you please provide a bit context why it is needed?
As far as I see you did not remove recovering on `PostConstruct` as well. So, now recovering it done twice, right?

Reply:
Ah, that's correct, the `recover()` in `@PostConstruct` should have been removed.